cpu/kinetis/periph/gpio.c

```diff
@@ -218,8 +218,21 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
     /* clear interrupt flags */
     port(pin)->ISFR &= ~(1 << pin_num(pin));
     /* enable global port interrupts in the NVIC */
+#ifdef KINETIS_CORE_Z
+    /* The Kinetis Cortex-M0+ parts usually only have two IRQs for GPIOs */
+    if (port_num(pin) == 0) {
+        /* PORTA gets its own IRQ */
+        NVIC_EnableIRQ(PORTA_IRQn);
+    }
+    else {
+        /* All other ports are combined on the next IRQ number */
+        NVIC_EnableIRQ(PORTA_IRQn + 1);
+    }
+#else
+    /* Cortex-M4 parts have one IRQ per port */
     NVIC_EnableIRQ(PORTA_IRQn + port_num(pin));
```

Comment:
all Kinetis header define `PORT_IRQS` so you could use a generic `NVIC_EnableIRQ(PORT_IRQS[port_num(pin)]);`, instead of this rather ugly `#ifdef` construct, or not? see https://github.com/RIOT-OS/RIOT/blob/master/cpu/kinetis/include/vendor/MKW41Z4.h#L5424 for instance

Reply:
I had never noticed that before, thanks! I will update accordingly

Reply:
😄 me neither, but as I quickly grep'ed through the code I stumbled over it and it look kind of useful and exactly what would fit here...